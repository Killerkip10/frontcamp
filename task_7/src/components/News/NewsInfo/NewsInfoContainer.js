import React from 'react';
import { connect } from 'react-redux';

import { NewsInfo } from './NewsInfo';
import { getNewsById } from './epicActions';

export const NewsInfoContainerComponent = ({ isFetching, info, getNewsById }) => (
  <NewsInfo
    isFetching={isFetching}
    info={info}
    getNewsById={getNewsById}
  />
);

const mapStateToProps = state => ({
  isFetching: state.news.newsInfo.isFetching,
  info: state.news.newsInfo.info,
});

const mapStateToDispatch = (dispatch, { match: { params } }) => ({
  getNewsById: topic => dispatch(getNewsById(params.index, params.topic)),
});

const connector = connect(mapStateToProps, mapStateToDispatch);

export const NewsInfoContainer = connector(NewsInfoContainerComponent);