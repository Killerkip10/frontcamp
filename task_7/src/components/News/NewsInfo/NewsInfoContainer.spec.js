import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';

import { NewsInfoContainerComponent } from './NewsInfoContainer';

describe('NewsInfoContainerComponent', () => {
    let props;

    beforeEach(() => {
        props = {
            isFetching: false,
            info: {
                subsection: 'subsection-1',
                title: 'title-1',
                abstract: 'abstract-1',
                section: 'section-1',
            },
            getNewsById: jest.fn(),
        };
    });

    it('should render NewsInfoContainerComponent', () => {
        const component = shallow(<NewsInfoContainerComponent {...props} />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render loader', () => {
        const component = shallow(<NewsInfoContainerComponent {...props} isFetching />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should call getNewsById', () => {
        mount(<NewsInfoContainerComponent {...props} />);

        expect(props.getNewsById).toBeCalled();
    });
});