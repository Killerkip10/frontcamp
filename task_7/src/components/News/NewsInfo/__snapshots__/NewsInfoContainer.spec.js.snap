// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsInfoContainerComponent should render NewsInfoContainerComponent 1`] = `
<Memo()
  info={
    Object {
      "abstract": "abstract-1",
      "section": "section-1",
      "subsection": "subsection-1",
      "title": "title-1",
    }
  }
/>
`;

exports[`NewsInfoContainerComponent should render loader 1`] = `
<div
  data-cy="loading"
>
  ...Loading
</div>
`;
